nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/shape/Concat.java

```diff
@@ -51,6 +51,13 @@ public Concat(int concatDimension, INDArray... arrays) {
         addIArgument(concatDimension);
     }
 
+    //TODO: This constructor to work around the code generator switching arguments.
+    public Concat(INDArray[] arrays, int concatDimension) {
+        super(null, arrays, new INDArray[0]);
```

Comment:
Actually I kind of prefer the "INDArrays first" approach, that's the convention we have elsewhere for op constructors. That said, varargs overload is kind of nice for usability though. Let's keep both in the op for now.
As for super - let's go with `super(arrays, null);` (yes, I know the existing op using INDArray[0], that should be changed to the same thing.

Reply:
updating both (in this branch)